salt/_modules/metalk8s_kubernetes_utils.py

```diff
@@ -40,7 +40,42 @@ def __virtual__():
     return __virtualname__
 
 
-def get_version_info(kubeconfig=None, context=None):
+def get_kubeconfig(**kwargs):
+    """
+    Get the kubeconfig and context from args or directly pillar or from
+    salt-master configuration.
+
+    Pillar value from `metalk8s.api_server.kubeconfig` and
+    `metalk8s.api_server.context`
+
+    Salt master config from `kubernetes.kubeconfig` and `kubernetes.context`
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt-call metalk8s_kubernetes.get_kubeconfig kubeconfig="/etc/kubernetes/admin.conf"
+        salt-call metalk8s_kubernetes.get_kubeconfig
+
+    Code Example:
+
+    .. code-block:: python
+
+        kubeconfig, context = __salt__['metalk8s_kubernetes.get_kubeconfig'](**kwargs)
+    """
+    pillar_dict = __pillar__.get('metalk8s', {}).get('api_server', {})
+
+    kubeconfig = kwargs.get('kubeconfig') or \
+        pillar_dict.get('kubeconfig') or \
+        __salt__['config.option']('kubernetes.kubeconfig')
```

Comment:
Don't you have access to `__opts__`? Or is that unrelated?

Reply:
We have access to `__opts__` just the initial `kubernetes` module from salt use this options so just kept it